app/validators/emoji_only_validator.rb

```diff
@@ -2,7 +2,7 @@ class EmojiOnlyValidator < ActiveModel::EachValidator
   DEFAULT_MESSAGE = "contains non-emoji characters or invalid emoji".freeze
 
   def validate_each(record, attribute, value)
-    return if value.gsub(EmojiRegex::RGIEmoji, "").blank?
+    return if !value || value.gsub(EmojiRegex::RGIEmoji, "").blank?
```

Comment:
this gsub acts on _all_ identified emoji - it's _possible_ for someone to insert an arbitrary number of emoji characters into a setting - this might not have been what was intended if we only expect a single emoji for the `Settings::Community.community_emoji` value?

Reply:
Minor comment: I'd find it a bit easier if we break this down into two guards:

```suggestion
    return unless value 
    return if value.gsub(EmojiRegex::RGIEmoji, "").blank?
```

Regarding your other concern, how about this?

```ruby
("\u{1F3ED}" * 20).match?(/\A#{EmojiRegex::RGIEmoji}\z/)
#=> false
"\u{1F3ED}".match?(/\A#{EmojiRegex::RGIEmoji}\z/)
#=> true
```

`gsub` doesn't seem like the right choice here anyway.

```suggestion
    return unless value 
    return if value.match?(/\A#{EmojiRegex::RGIEmoji}\z/)
```

Reply:
The change to the matcher here means empty strings are no longer valid (which is not a problem if the first guard becomes `unless value.present?`). 

```ruby
valid1 = "\u{1F3ED}"
=> "🏭"
valid2 = ""
invalid = valid1 * 2
=> "🏭🏭"
valid1.match?(/\A#{EmojiRegex::RGIEmoji}\z/)
=> true
valid2.match?(/\A#{EmojiRegex::RGIEmoji}\z/)
=> false
invalid.match?(/\A#{EmojiRegex::RGIEmoji}\z/)
=> false
```

I'm not opposed to improving this/tightening it up - it's a little outside of the scope of what I set out to do initially (resolve a user facing error).

Reply:
Agreed, it's outside the scope of this PR. Let's fix the actual problem first, then get some clarity around the original intention.